client/js/lounge.js

```diff
@@ -45,14 +45,24 @@ $(function() {
 
 	// Autocompletion Strategies
 
+	const emojiSearchTerms = Object.keys(emojiMap);
 	const emojiStrategy = {
 		id: "emoji",
 		match: /\B:([-+\w]*):?$/,
 		search(term, callback) {
-			callback(Object.keys(emojiMap).filter((name) => name.indexOf(term) === 0));
+			const results = fuzzy.filter(
+				term,
+				emojiSearchTerms,
+				{
+					pre: "<b>",
+					post: "</b>"
+				}
+			);
+			callback(results.map((el) => el.string));
 		},
 		template(value) {
-			return `<span class="emoji">${emojiMap[value]}</span> ${value}`;
+			const cleansed = value.replace(/<b>|<\/b>/g, "");
+			return `<span class="emoji">${emojiMap[cleansed]}</span> ${value}`;
```

Comment:
So, I don't like that we add the boldness in one method, then remove it in the next one, especially since `fuzzy` has an `original` key. I played a bit locally, and made this step go away with the following (and some nice ES6 destructured assignments):

```diff
─────────────────────────
modified: client/js/lounge.js
─────────────────────────
@ lounge.js:61 @ $(function() {
                    post: "</b>"
                }
            );
-           callback(results.map((el) => el.string));
+           callback(results.map((el) => [el.string, el.original]));
        },
-       template(value) {
-           const cleansed = value.replace(/<b>|<\/b>/g, "");
-           return `<span class="emoji">${emojiMap[cleansed]}</span> ${value}`;
+       template([string, original]) {
+           return `<span class="emoji">${emojiMap[original]}</span> ${string}`;
        },
-       replace(value) {
-           return emojiMap[value];
+       replace([, original]) {
+           return emojiMap[original];
        },
        index: 1
    };
```